lib/node_modules/@stdlib/types/index.d.ts

```diff
@@ -593,6 +593,85 @@ declare module '@stdlib/types/array' {
 	* const x: Collection<number> = [ 1, 2, 3 ];
 	*/
 	type Collection<T = any> = Array<T> | TypedArray | ArrayLike<T>; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `float64`: double-precision floating-point numbers (IEEE 754)
+	* -   `float32`: single-precision floating-point numbers (IEEE 754)
+	*/
+	type RealFloatingPointArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'float64': Float64Array;
+		'float32': Float32Array;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `complex128`: double-precision complex floating-point numbers
+	* -   `complex64`: single-precision complex floating-point numbers
+	*/
+	type ComplexArrayMap = { // eslint-disable-line @typescript-eslint/consistent-type-definitions
+		'complex64': Complex64Array;
+		'complex128': Complex128Array;
+	};
+
+	/**
+	* Type mapping for the following array data types:
+	*
+	* -   `complex64`: single-precision complex floating-point numbers
```

Comment:
This is a decent example of the "drift" I mention above. If the `*Map` speaks for itself, I think we can remove these lists.

Also, from the description not clear what the mapping is for. So, instead, maybe

> Mapping of integer typed array data types to integer typed array constructors.

Reply:
Agreed, let's do that.